cloudpickle/cloudpickle.py

```diff
@@ -403,15 +403,18 @@ def save_dynamic_class(self, obj):
         from global modules.
         """
         clsdict = dict(obj.__dict__)  # copy dict proxy to a dict
-        if not isinstance(clsdict.get('__dict__', None), property):
-            # don't extract dict that are properties
-            clsdict.pop('__dict__', None)
-            clsdict.pop('__weakref__', None)
+        clsdict.pop('__weakref__', None)
```

Comment:
This now unconditionally pops `__weakref__`, since it's not meaningful to pickle it.